autorest/codegen/templates/operations_container_init.py.jinja2

```diff
@@ -7,7 +7,7 @@ from .{{ ("_operations" if code_model.options['combine_operation_files'] else op
 # coding=utf-8
 {{ code_model.options['license_header'] }}
 
-{% if not code_model.options["add_python_3_operation_files"] %}
+{% if (not code_model.options["add_python_3_operation_files"]) or async_mode %}
```

Comment:
good catch, thanks!

small nit: you don't need the parentheses around `not code_model.options["add_python_3_operation_files"]`